radixdlt/radixdlt-scrypto
# feature: Added transaction scenarios framework

## Summary

Added a transaction scenarios framework, which can be used:
* To make testnet ledgers more interesting / comprehensive, to aid in testing of Core API / Gateway API, and for Dashboard
* To provide many _valid_ example manifests for reference by integrators
* To allow for more comprehensive testing of the 

which has three key goals:
* Test all the different substates
  * Provide test data covering for testing the Core API substate mapping
* Create a more interesting ledger for testnets
* Provide a swathe of _valid_ example manifests for more complicated journeys, auto-generated
  * Future work: we could rebuild the manifest value formatter on top of a traversal to get it to auto-output the enums as eg `Well::Known` - which we don't have time for right now)

Still to come on a follow-up PR (maybe for Cedar as we're kinda too late for birch now...)
* Create lots more scenarios - they should be really easy to add now I hope!
* Adding all the scenarios to the `test_bootstrap_receipt_should_have_substate_changes_which_can_be_typed` test in `radix-engine-tests`.
* Create a ManifestBuilder2 which uses self, and allocator
